src/track/taglib/trackmetadata_file.cpp

```diff
@@ -114,20 +114,24 @@ bool hasMP4Tag(TagLib::MP4::File& file) {
 bool readAudioPropertiesFromFile(
         TrackMetadata* pTrackMetadata,
         const TagLib::File& file) {
-    // The declaration of TagLib::FileName is platform specific.
+    if (!file.isValid()) {
+        // The declaration of TagLib::FileName is platform specific.
 #ifdef _WIN32
-    // For _WIN32 there are two types std::string and std::wstring
-    // we must pick one explicit,
-    // to prevent "use of overloaded operator '<<' is ambiguous" error
-    // on clang-cl builds.
-    const std::wstring& filename = file.name().wstr();
+        // For _WIN32 there are two types std::string and std::wstring
+        // we must pick one explicit,
+        // to prevent "use of overloaded operator '<<' is ambiguous" error
+        // on clang-cl builds.
+        TagLib::FileName filename = file.name();
```

Comment:
I think the best way to fix this would be something like this:
```cpp
#ifdef _WIN32
TagLib::FileName filename_owning = file.name();
auto filename = std::basic_string_view(filename_owning.wstr());
#else 
auto filename = std::basic_string_view(file.name());
#endif
```
this is polymorphic over the CharT thanks to CTAD and does not involve conversion. Could be even simpler if we use just convert to `char*` on windows directly, or was there a reason to use the wide variant?

Reply:
I wonder what implications the unfortunate API design has for ownership? Eg is `file.name()` just leaking the string if not on windows?

Reply:
file.name() is just a non owning pointer to the original string on Linux/MacOS
I have kept the native types for filename to have this issue more visible.

Reply:
okay, so no leaks, just extra memory allocation on windows it seems.